raiden/transfer/mediated_transfer/mediator.py

```diff
@@ -243,8 +256,11 @@ def sanity_check(state):
         assert is_send_transfer_almost_equal(transfer_sent, transfer_received)
 
 
-def clear_if_finalized(iteration, channelidentifiers_to_channels):
-    """ Clear the state if all transfer pairs have finalized. """
+def clear_if_finalized(
+        iteration: TransitionResult,
+        channelidentifiers_to_channels: typing.ChannelMap,
+) -> TransitionResult:
+    """ Clear the state if all transfer pairs have finalized or all channels are removed """
```

Comment:
This docstring is incorrect IMO.

It is not about having either all pairs finalized *or* all channels removed. It's about having for both payer and payee locks from each pair finalized, which may be done offchain or onchain, and onchain it will be with a channel removal.

What about this?

```
Clear the mediator task if all the locks have been finalized.

A lock is considered finalized if it has been removed from the merkle tree
offchain, either because the transfer was unlocked or expired, or because the
channel was settled on chain and therefore the channel it's removed. 
```

Reply:
Thanks this sounds better indeed.